client/webserver/site/src/js/orders.js

```diff
@@ -179,6 +164,29 @@ export default class OrdersPage extends BasePage {
     return res.orders
   }
 
+  /* exportOrders opens a new page that downloads a csv of orders based on the
+   * current filters */
+  async exportOrders () {
+    const currentFilter = { ...this.currentFilter() }
+    // return all orders instead of only the ones on the current page
+    currentFilter.n = 0
+    currentFilter.offset = ''
+    const res = await postJSON('/api/orderscsv', this.currentFilter())
+    if (res.ok) {
+      // eslint-disable-next-line no-undef
+      const csvFile = new Blob([res.csv], { type: 'text/csv' })
```

Comment:
A POST request was needed to be able to send a JSON object which could be deserialized as an OrderFilter. To trigger a download after a POST request, a fake link has to be created and clicked programatically. It's a bit hacky, but it seems to be the way most people do this.